core/chains/evm/logpoller/log_poller.go

```diff
@@ -375,15 +377,16 @@ func (lp *logPoller) pollAndSaveLogs(ctx context.Context, currentBlockNumber int
 	// E.g. 1<-2<-3(currentBlockNumber)<-4<-5<-6<-7(latestBlockNumber), finality is 2. So 3,4 can be batched.
 	// Although 5 is finalized, we still need to save it to the db for reorg detection if 6 is a reorg.
 	// start = currentBlockNumber = 3, end = latestBlockNumber - finality - 1 = 7-2-1 = 4 (inclusive range).
-	if (latestBlockNumber - currentBlockNumber) >= (lp.finalityDepth + 1) {
-		lp.lggr.Infow("Backfilling logs", "start", currentBlockNumber, "end", latestBlockNumber-lp.finalityDepth-1)
-		currentBlockNumber = lp.backfill(ctx, currentBlockNumber, latestBlockNumber-lp.finalityDepth-1)
+	lastSafeBackfillBlock := latestBlockNumber - lp.finalityDepth - 1
```

Comment:
I feel like using the same math as in the comment explaining it helps with readability